devito/types/basic.py

```diff
@@ -1147,13 +1152,16 @@ def _evaluate(self, **kwargs):
             return self
 
         # Base function
-        retval = self.function
+        retval = 1 / self.function if self._avg_mode == 'harmonic' else self.function
```

Comment:
nitpicking: for homogeneity with the rest of the codebase, it should be

```
if cond:
     <then part>
else:
     <else part>
```

even if it's a trivial code fragment. But again, nitpicking...